src/generate/register.rs

```diff
@@ -636,14 +636,23 @@ impl Variant {
 fn add_from_variants(mod_items: &mut Vec<TokenStream>, variants: &Vec<Variant>, pc: &Ident, f: &F, desc: &str, reset_value: Option<u64>) {
     let fty = &f.ty;
 
+    let mut repr = quote! { #[repr(#fty)] };
+    let mut cast = quote! { variant as _ };
+
+    if f.ty == "bool" {
+        repr = quote! { };
+        cast = quote! { variant as u8 != 0 };
+    }
```

Comment:
Nitpick: This could be if else instead, have no real preference however.

Reply:
Done